tests/aws/services/s3/test_s3_list_operations.py

```diff
@@ -46,6 +48,87 @@ def assert_timestamp_is_iso8061_s3_format(timestamp: str):
     assert parsed_ts.microsecond == 0
 
 
+class TestS3ListBuckets:
+    @markers.aws.validated
+    def test_list_buckets_by_prefix_with_case_sensitivity(
+        self, s3_create_bucket, aws_client, snapshot
+    ):
+        snapshot.add_transformer(snapshot.transform.s3_api())
+
+        bucket_name = f"test.bucket.{short_uid()}"
+        s3_create_bucket(Bucket=bucket_name)
+        s3_create_bucket(Bucket=f"ignored.bucket.{short_uid()}")
+
+        response = aws_client.s3.list_buckets(Prefix=bucket_name.upper())
+        assert len(response["Buckets"]) == 0
+
+        response = aws_client.s3.list_buckets(Prefix=bucket_name)
+        assert len(response["Buckets"]) == 1
+
+        returned_bucket = response["Buckets"][0]
+        assert returned_bucket["Name"] == bucket_name
+
+        snapshot.match("list-objects-by-prefix", response)
+
+    @markers.aws.validated
+    @markers.snapshot.skip_snapshot_verify(paths=["$..ContinuationToken"])
+    def test_list_buckets_with_max_buckets(self, s3_create_bucket, aws_client, snapshot):
+        snapshot.add_transformer(snapshot.transform.s3_api())
+
+        s3_create_bucket()
+        s3_create_bucket()
+
+        response = aws_client.s3.list_buckets(MaxBuckets=1)
+        assert len(response["Buckets"]) == 1
+
+        snapshot.match("list-objects-with-max-buckets", response)
+
+    @markers.aws.validated
+    @markers.snapshot.skip_snapshot_verify(paths=["$..ContinuationToken"])
+    def test_list_buckets_by_bucket_region(
+        self, s3_create_bucket, s3_create_bucket_with_client, aws_client_factory, snapshot
+    ):
+        region_us_west_2 = "us-west-2"
+        snapshot.add_transformer(snapshot.transform.s3_api())
+        snapshot.add_transformer(RegexTransformer(region_us_west_2, "<region>"))
+
+        s3_create_bucket()
+
+        client_us_west_2 = aws_client_factory(region_name="us-west-2").s3
+        s3_create_bucket_with_client(
+            client_us_west_2,
+            Bucket=f"test-bucket-{short_uid()}",
+            CreateBucketConfiguration={"LocationConstraint": region_us_west_2},
+        )
+
+        response = client_us_west_2.list_buckets(BucketRegion=region_us_west_2)
+
+        returned_bucket = response["Buckets"][0]
+        assert returned_bucket["BucketRegion"] == region_us_west_2
+
+        snapshot.match("list-objects-by-bucket-region", response)
```

Comment:
nit: might be good to add a case where we try listing buckets in another region, and for it to return an empty result?

Reply:
I think we could then update the test to use both the `BucketRegion` and the `Prefix` to make sure to return only the bucket you want, and then use the wrong region and the `Prefix`? 

I believe the issue you're mentioning could also be happening in this case if anything creates a bucket in `us-west-2`.

I think those list operations on global resources like this one are indeed tricky for parallelization, and most of those tests would fail when asserting the length of the list response.